staff/edit-courses/edit-course.js

```diff
@@ -0,0 +1,359 @@
+    }, 1, btn);
+}
+
+function dropdownSelect(selection) {
+    timeout = setTimeout(function(selection) {
+        let drop = selection.parentNode.previousElementSibling;
+        let choiceValue = selection.attributes.value.nodeValue;
+        let choiceText = selection.text;
+        drop.value = choiceValue;
+        drop.innerText = choiceText;
+
+        // Hide or show year box if altering a requisite semester
+        if (drop.id == "req-sem") {
+            if (drop.value == 'none') {
+                drop.nextElementSibling.nextElementSibling.style.display='none';
+            } else {
+                drop.nextElementSibling.nextElementSibling.style.display='';
+            }
+        }
+
+    }, 1, selection);
+}
+
+function addReq() {
+    let table = document.getElementById("reqs-table");
+    table.insertRow();
+
+    let current_req = nextReq;
+    fetch("requisite.php?req_num="+current_req).then(
+        response => response.text()
+    ).then(function(text) {
+        table.rows[table.rows.length -1].outerHTML = text;
+        updateReqAutoComplete(current_req);
+    });
+    nextReq++;
+}
+
+function removeReq(btn) {
+    let row = btn.parentNode.parentNode;
+    row.parentNode.removeChild(row);
+}
+
+function populateModal(btn) {
+    let course_code = "";
+    
+    if (btn.innerText.trim() === "Add new course") {
+        course_code = "New";
+    } else {
+        let course_row = btn.parentNode.parentNode;
+        course_code = course_row.children[0].innerText;
+    }
+
+    fetch("edit-course.php?course_code="+course_code).then(
+        response => response.text()
+    ).then(
+        function(text) {
+            document.getElementById("edit-course").innerHTML = text;
+            // Find last req in course to determine how many need updated with autocomplete
+            let req_pos = text.lastIndexOf("reqCode");
+            let req_end = text.indexOf("'", req_pos);
+            let req_id = text.slice(req_pos + "reqCode-".length, req_end);
+            for (let i = 0; i <= req_id; i++) {
+                // Update all reqs with autocomplete functionality
+                updateReqAutoComplete(i);
+            }
+            // Handle submit button manually to prevent enter from pressing default button
+            $('#form-button-submit').click(function(){
+                $('#edit-course-form').submit();
+              });
+            document.getElementById("course_code").addEventListener('change', function() {
+                document.getElementById("course_code").style.border = "";
+            });
+    });
+}
+
+// Set a hidden input following a button to the state of the button
+// So that the value is included in the form POST
+function addPost(btn) {
+    timeout = setTimeout(function(btn) {
+        let input = btn.nextElementSibling;
+        input.value = btn.ariaPressed;
+    }, 3, btn); // wait 3 to ensure this occurs after the value is changed
+}
+
+function to_Semester(year, season) {
```

Comment:
I wonder if we should make some semester functions like this common.js since I also have some of this in Semester.js